pkg/descheduler/strategies/duplicates.go

```diff
@@ -131,7 +143,14 @@ func RemoveDuplicatePods(
 				for _, keys := range existing {
 					if reflect.DeepEqual(keys, podContainerKeys) {
 						matched = true
-						duplicatePods = append(duplicatePods, pod)
+						klog.V(3).InfoS("Duplicate found", "pod", klog.KObj(pod))
+						for _, ownerRef := range ownerRefList {
+							ownerKey := podOwner{pod.ObjectMeta.Namespace, ownerRef.Kind, ownerRef.Name}
+							if _, ok := duplicatePods[ownerKey]; !ok {
+								duplicatePods[ownerKey] = make(map[string][]*v1.Pod)
+							}
+							duplicatePods[ownerKey][node.Name] = append(duplicatePods[ownerKey][node.Name], pod)
```

Comment:
will this work if 2 different pods have the same owner?

Reply:
To show what I mean, see the PR I opened here (picked your branch and added a test case that fails): https://github.com/kubernetes-sigs/descheduler/pull/467/commits/71a2a7e9e6556579cca4bf29b584797251db9c79

say we have 4 pods with the same ownerref, but only 2 of them are actually duplicates. with how you currently count occurrences, the average gets calculated out to 2 for this ownerKey, but it should be 1 because ownerKey counts are unique to the image

I think we can still fix up this PR, but we need to refactor how you're counting the occurrences of each, perhaps to just include the image? Something like that could fix it

Reply:
Nice catch!!! PR updated.